src/components/ContentHighlights/HighlightStepper/HighlightStepperTitleInput.jsx

```diff
@@ -11,37 +11,59 @@ const HighlightStepperTitleInput = () => {
   const highlightTitle = useContextSelector(ContentHighlightsContext, v => v[0].stepperModal.highlightTitle);
   const [titleLength, setTitleLength] = useState(highlightTitle?.length || 0);
   const [isInvalid, setIsInvalid] = useState(false);
-
-  const handleChange = (e) => {
+  const [highlightValue, setHighlightValue] = useState({
+    initialized: false,
+    highlightTitle: highlightTitle || '',
+    titleStepValidationError: undefined,
+    highlightTitleLength: titleLength || 0,
+  });
+  const handleChange = ((e) => {
+    e.persist();
     if (e.target.value.length > MAX_HIGHLIGHT_TITLE_LENGTH) {
       setIsInvalid(true);
-      setHighlightTitle({
+      setHighlightValue({
+        initialized: true,
         highlightTitle: e.target.value,
         titleStepValidationError: DEFAULT_ERROR_MESSAGE.EXCEEDS_HIGHLIGHT_TITLE_LENGTH,
+        highlightTitleLength: e.target.value.length,
       });
     } else {
       setIsInvalid(false);
-      setHighlightTitle({
+      setHighlightValue({
+        initialized: true,
         highlightTitle: e.target.value,
         titleStepValidationError: undefined,
+        highlightTitleLength: e.target.value.length,
       });
     }
-    setTitleLength(e.target.value.length);
-  };
+  });
 
+  useEffect(() => {
+    if (highlightValue.initialized) {
+      setHighlightTitle({
+        highlightTitle: highlightValue.highlightTitle,
+        titleStepValidationError: highlightValue.titleStepValidationError,
+      });
+      setTitleLength(highlightValue.highlightTitleLength);
+      setHighlightValue({
+        ...highlightValue,
```

Comment:
note: this is relying on the actual state variable in a state setter function. As such, there's no guarantee that `highlightValue` will be the correct/intended reference since the `useEffect` is re-triggered whenever the `highlightValue` itself changes.

Recommended to use the callback function approach to using the previous state to remove it from the dependency array of the `useEffect`:

```jsx
setHighlightValue(prevState = ({
 ...prevState,
  initialized: false,
}));
```